src/structures/lists/digest.rs

```diff
@@ -1,11 +1,72 @@
 // Copyright 2020 Contributors to the Parsec project.
 // SPDX-License-Identifier: Apache-2.0
+use crate::constants::algorithm::HashingAlgorithm;
 use crate::structures::Digest;
-use crate::tss2_esys::TPML_DIGEST;
+use crate::tss2_esys::{TPML_DIGEST, TPML_DIGEST_VALUES, TPMT_HA, TPMU_HA};
 use crate::{Error, Result, WrapperErrorKind};
 use log::error;
+use std::collections::HashMap;
 use std::convert::{TryFrom, TryInto};
 
+#[derive(Debug, Clone, Default)]
+pub struct DigestValues {
+    digests: HashMap<HashingAlgorithm, Digest>,
+}
+
+impl DigestValues {
+    pub fn new() -> Self {
+        DigestValues {
+            digests: HashMap::new(),
+        }
+    }
+
+    pub fn set(&mut self, alg: HashingAlgorithm, dig: Digest) {
+        let _ = self.digests.insert(alg, dig);
+    }
+}
+
+impl TryFrom<(HashingAlgorithm, Digest)> for TPMT_HA {
+    type Error = Error;
```

Comment:
I like this part but I think it should be placed in a separate module.

In ```src/structures``` is where we have placed most of our rust representations of the TPM types (i.g. Name which is our way of representing TPM2B_NAME can be found in ```src/structures/names/name.rs```). So I would suggest you do ```src/structures/hash_digest.rs``` and put in your code there for your hash digest there.

Reply:
Thanks for the note. My intention had indeed been to split `DigestValues` out, but I forgot to do that before committing!